### ISSUE TITLE
resource_keycloak_openid_client_authorization_permission missing resource_type field 

### ISSUE THREAD

**Issue by jhinzmann** [mrparkers/terraform-provider-keycloak#398]
Currently the provider for `resource_keycloak_openid_client_authorization_permission` accepts a list of resources and/or scopes. 
Besides the list of specific resources or scopes, the Keycloak UI allows to create a resource based permission that is applied to all resources of a specific type.

The corresponding payload for creating a permission for all resources of type `test-resource-type` looks like this:
```json
{
  "type":"resource",
  "logic":"POSITIVE",
  "decisionStrategy":"UNANIMOUS",
  "name":"test-permission",
  "description":"test-description",
  "resourceType":"test-resource-type",
  "policies":[
    "3d13b164-f4e1-4606-8f87-1bfa8b099919"
  ]
}
```

Therefore I propose to support a new optional field `resource_type` that allows to set the string value for the `resourceType` of the permission.

**Comment by JessieAMorris** [comment]
@mrparkers I've opened a PR for this issue. This is blocking me from implementing a feature on my project and would love get this moved forward.

**Comment by mrparkers** [comment]
This has been released in v3.10.0, thanks everyone!
